paasta_tools/marathon_tools.py

```diff
@@ -674,25 +676,53 @@ def format_marathon_app_dict(self) -> FormattedMarathonAppDict:
         code_sha = get_code_sha_from_dockerurl(docker_url)
 
         config_hash = get_config_hash(
-            self.sanitize_for_config_hash(complete_config),
+            self.sanitize_for_config_hash(complete_config, system_paasta_config),
             force_bounce=self.get_force_bounce(),
         )
         complete_config['id'] = format_job_id(self.service, self.instance, code_sha, config_hash)
 
         log.debug("Complete configuration for instance is: %s", complete_config)
         return complete_config
 
-    def sanitize_for_config_hash(self, config: FormattedMarathonAppDict) -> Dict[str, Any]:
+    def sanitize_for_config_hash(
+        self,
+        config: FormattedMarathonAppDict,
+        system_paasta_config: SystemPaastaConfig,
+    ) -> Dict[str, Any]:
         """Removes some data from complete_config to make it suitable for
         calculation of config hash.
 
+        Also adds secret HMACs so that we bounce if secret data has changed.
+        We need this because the reference to the secret is all Marathon gets
+        and this will not change.
+
         :param config: complete_config hash to sanitize
         :returns: sanitized copy of complete_config hash
         """
         ahash = {key: copy.deepcopy(value) for key, value in config.items() if key not in CONFIG_HASH_BLACKLIST}
         ahash['container']['docker']['parameters'] = self.format_docker_parameters(with_labels=False)  # type: ignore
+        secret_hashes = self.get_secret_hashes(config['env'], system_paasta_config)
+        if secret_hashes:
```

Comment:
this is the secret sauce thats going to avoid a big bounce right?

Reply:
Right, that's the idea